### DISCUSSION TITLE
Can we use the referee.py in a local environment?

### DISCUSSION THREAD

**Discussion by YuriCat** [cyberbotics/wrestling#108] [Q&A] [answered]
Dear all,

I think we need a referee.py to determine the winner and loser between agents in our local environments.
Just running the simulation in docker did not give me a winner, and the simulation continued beyond 3 minutes.

Does anyone know if this is possible?

**Comment by omichel** [comment]
You should run the simulation locally, but not inside a Docker and the supervisor should behave properly.

**Reply by YuriCat** [reply]
Thank you for your immediate response.
I can confirm that the wins and losses are displayed on the webots GUI.

For example, if I want to run 100 matches (in parallel if possible) and get the wins/losses, I would like to run it on the command line or otherwise record the wins/losses for each match and start the next match immediately.

In other words, I would like to know how to build an evaluation environment.

**Reply by omichel** [reply, marked as answer]
In that case, you may have to write some scripts that launch Webots and maybe also to modify the supervisor controller so that it stores the outcome of a match in a file and quits Webots.

**Reply by YuriCat** [reply]
I see, so you are saying that webots just do a simulation, and cannot return the results and finish the game, so it will be prepared externally. I understand.
I guess we can do that.
Thank you very much!
